test/test_torch.py

```diff
@@ -17836,49 +17836,68 @@ def test_case_info(fn_name, config):
                         f'Subprocess exception while attempting to run {test_case_info(fn_name, config)}:\n'
                         + e.output.decode("utf-8")))
 
-    @onlyCPU
+    @precisionOverride({torch.half: 0.05, torch.bfloat16: 0.05})
+    @onlyOnCPUAndCUDA
+    @dtypesIfCUDA(*(torch.testing.get_all_fp_dtypes(include_bfloat16=AMPERE_OR_ROCM)))
     @dtypes(*(torch.testing.get_all_complex_dtypes() + [torch.float, torch.double]))
+    @tf32_on_and_off(0.05)
     def test_addbmm(self, device, dtype):
         # num_batches = 10
         # M, N, O = 12, 8, 5
         num_batches = 2
         M, N, O = 2, 3, 4
-        b1 = torch.randn(num_batches, M, N, dtype=dtype, device=device)
-        b2 = torch.randn(num_batches, N, O, dtype=dtype, device=device)
-        res = torch.bmm(b1, b2)
-        res2 = torch.tensor((), dtype=dtype, device=device).resize_as_(res[0]).zero_()
-        res3 = torch.tensor((), dtype=dtype, device=device).resize_as_(res[0]).zero_()
-
-        res2.addbmm_(b1, b2)
-        self.assertEqual(res2, res.sum(0, False))
-        res3.copy_(res2)
-
-        with self.maybeWarnsRegex(
-                UserWarning, "This overload of addbmm_ is deprecated"):
-            res2.addbmm_(1, b1, b2)
-        self.assertEqual(res2, res.sum(0, False) * 2),
-        res3.addbmm_(b1, b2, beta=1)
-        self.assertEqual(res2, res3)
-
-        with self.maybeWarnsRegex(
-                UserWarning, "This overload of addbmm_ is deprecated"):
-            res2.addbmm_(1., .5, b1, b2)
-        self.assertEqual(res2, res.sum(0, False) * 2.5)
-        res3.addbmm_(b1, b2, beta=1., alpha=.5)
-        self.assertEqual(res2, res3)
-
-        with self.maybeWarnsRegex(
-                UserWarning, "This overload of addbmm is deprecated"):
-            self.assertEqual(res2, torch.addbmm(1, res2, 0, b1, b2))
-
-        res4 = torch.addbmm(res2, b1, b2, beta=1, alpha=.5)
-        self.assertEqual(res4, res.sum(0, False) * 3),
-
-        res5 = torch.addbmm(res2, b1, b2, beta=0, alpha=1)
-        self.assertEqual(res5, res.sum(0, False))
 
-        res6 = torch.addbmm(res2, b1, b2, beta=.1, alpha=.5)
-        self.assertEqual(res6, res2 * .1 + .5 * res.sum(0)),
+        def invert_perm(p):
+            d = {x: i for i, x in enumerate(p)}
+            return (d[0], d[1], d[2])
+
+        for perm1, perm2 in product(permutations((0, 1, 2)), repeat=2):
+            for perm3 in permutations((0, 1)):
+                b1 = torch.randn(num_batches, M, N, dtype=dtype, device=device)
```

Comment:
same as for baddbmm, please compare with numpy instead of this convoluted test logic. Also, if it makes sense to unify this test with baddbmm (at first glance it looks similar), please do so.

Reply:
I think it's better to test separately. The reason is, the output of `addbmm` is 2D but `baddbmm` is 3D, so it is
```python
        for perm1, perm2 in product(permutations((0, 1, 2)), repeat=2):
            for perm3 in permutations((0, 1)):
```
vs
```python
for perm1, perm2, perm3 in product(permutations((0, 1, 2)), repeat=3):
```
I don't know how these two can be combined easily.

Reply:
Also, I don't think numpy has something like `addbmm`, the the difference is only `torch.bmm(b1, b2).sum(0, False)` vs `numpy.matmul(b1, b2).sum(0, False)`, in this case, does it still make sense to change to numpy?